pkg/e2e/framework/framework.go

```diff
@@ -59,83 +59,122 @@ func init() {
 	flag.StringVar(&sshuser, "ssh-user", "ec2-user", "Ssh user to connect to instances")
 	flag.StringVar(&actuatorImage, "actuator-image", "gcr.io/k8s-cluster-api/machine-controller:0.0.1", "Actuator image to run")
 
+	flag.StringVar(&proxySSHUser, "proxy-ssh-user", "root", "Ssh user for proxy")
+	flag.StringVar(&proxySSHKey, "proxy-ssh-key", "", "Ssh key for proxy")
+	flag.StringVar(&proxySSHHost, "proxy-ssh-host", "", "Ssh host for proxy")
+
 	flag.Parse()
 }
 
+type ErrNotExpectedFnc func(error)
+type ByFnc func(string)
+
+type SSHConfig struct {
+	Key  string
+	User string
+	Host string
+}
+
 // Framework supports common operations used by tests
 type Framework struct {
 	KubeClient            *kubernetes.Clientset
 	CAPIClient            *clientset.Clientset
 	APIRegistrationClient *apiregistrationclientset.Clientset
 	Kubeconfig            string
 	RestConfig            *rest.Config
-	SSHKey                string
-	SSHUser               string
-	ActuatorImage         string
+
+	SSH *SSHConfig
+
+	ActuatorImage  string
+	ErrNotExpected ErrNotExpectedFnc
+	By             ByFnc
 }
 
 // NewFramework setups a new framework
 func NewFramework() *Framework {
```

Comment:
Make the name include `Must` - gives a clue it can panic. Alternatively, return an error.

Reply:
Example? `NewMustFramework`?

Reply:
Yeah, kind of awkward. MustCreateFramework? Dunno... /sorry.

Reply:
I think leading `Must` helps.

Reply:
Returning error instead